src/main/webapp/app/overview/course-dashboard/course-dashboard.component.ts

```diff
@@ -0,0 +1,190 @@
+            case ScoreType.PRESENTATION_SCORE:
+                return scores.studentScores.presentationScore;
+            case ScoreType.REACHABLE_PRESENTATION_POINTS:
+                return scores.reachablePresentationPoints;
+        }
+    }
+    ngOnDestroy(): void {
+        this.paramSubscription?.unsubscribe();
+        this.courseUpdatesSubscription?.unsubscribe();
+    }
+
+    /**
+     * Loads all prerequisites and competencies for the course
+     */
+    loadCompetencies() {
+        this.isLoading = true;
+        forkJoin([this.competencyService.getAllForCourse(this.courseId), this.competencyService.getAllPrerequisitesForCourse(this.courseId)]).subscribe({
+            next: ([competencies, prerequisites]) => {
+                this.competencies = competencies.body!;
+                this.prerequisites = prerequisites.body!;
+                // Also update the course, so we do not need to fetch again next time
+                if (this.course) {
+                    this.course.competencies = this.competencies;
+                    this.course.prerequisites = this.prerequisites;
+                }
+                this.isLoading = false;
+            },
+            error: (errorResponse: HttpErrorResponse) => onError(this.alertService, errorResponse),
+        });
+    }
+
+    get nonMasteredCompetencies(): Competency[] {
+        return this.competencies.filter((competency) => {
+            if (competency.userProgress?.length && competency.masteryThreshold) {
+                return competency.userProgress.first()!.progress! < 100 || competency.userProgress.first()!.confidence! < competency.masteryThreshold!;
+            }
+            return false;
+        });
+    }
+
+    get nextCompetency(): Competency | undefined {
+        if (this.nonMasteredCompetencies.length > 0) {
+            return this.nonMasteredCompetencies[0];
+        }
+        return this.competencies.at(-1);
+    }
+
+    get nextExercise() {
+        return this.nextCompetency?.exercises?.find((exercise) => exercise.dueDate && exercise.dueDate.isAfter(dayjs()) && !this.filteredExerciseIDs.includes(exercise.id!));
+    }
+
+    get countCompetencies(): number {
+        return this.competencies.length;
+    }
+
+    private setCourse(course?: Course) {
+        this.course = course;
+        this.onCourseLoad();
+        // Note: this component is only shown if there are at least 1 competencies or at least 1 prerequisites, so if they do not exist, we load the data from the server
+        if (this.course && ((this.course.competencies && this.course.competencies.length > 0) || (this.course.prerequisites && this.course.prerequisites.length > 0))) {
+            this.competencies = this.course.competencies || [];
+            this.prerequisites = this.course.prerequisites || [];
+        } else {
+            this.loadCompetencies();
+        }
+        this.data = {
+            course: this.course,
+        };
+    }
```

Comment:
Can you also remove the unnecessary parts from here? We are not using the competencies exercise statistics etc. at the moment. I would add them with the PR that actually makes use of them.